spec.html

```diff
@@ -28965,13 +28966,27 @@ <h1>
         <h1>
           HostEnsureCanCompileStrings (
             _calleeRealm_: a Realm Record,
+            _goal_: a grammar symbol,
+            _bodyText_: a String,
           ): either a normal completion containing ~unused~ or a throw completion
         </h1>
         <dl class="header">
           <dt>description</dt>
           <dd>It allows host environments to block certain ECMAScript functions which allow developers to interpret and evaluate strings as ECMAScript code.</dd>
         </dl>
         <p>The default implementation of HostEnsureCanCompileStrings is to return NormalCompletion(~unused~).</p>
+
+        <emu-note>
+          <p>
+            _goal_ is the production which will be used to parse _bodyText_.
+            For example, if called via %Function% it might be the grammar symbol |FunctionBody[~Yield, ~Await]| and from %eval% it might be the grammar symbol |Script|.
+          </p>
+        </emu-note>
+
+        <p>
+          Implementations of HostEnsureCanCompileStrings should not coerce _bodyText_ to a string since callers will later do that and HostEnsureCanCompileStrings must not base security decisions on both coercions producing the same string.
+          Clients may base security decisions on internal slots.
```

Comment:
what does this mean?

Reply:
This note I included as it was in #1498 so maybe @mikesamuel can correct me if I'm wrong. My understanding is that if you try to compile an object as code (which cannot happen through `eval(myObject)`, but can happen through `new Function('param', myObject)` it's valid for the host to use `myObject`'s internal slots to decide whether to block this operation or not.

Reply:
```suggestion
          If _bodyText_ is an Object, the host may examine its internal slots.
```

Reply:
Does that need to be specified? Can't hosts make that decision based on any criteria they want?

Reply:
Yeah, I guess it can be removed, as it's implicitly allowed.

I assume this line was originally intended to emphasize that if _bodyText_ is a [`TrustedScript`](https://w3c.github.io/trusted-types/dist/spec/#trusted-script), the host should not call its `toString()` method but instead examine its [[Data]] internal slot.